### PR TITLE
Site breakage report screen improvements

### PR DESCRIPTION
**Pull request by lmac012**
Task/Issue URL: https://app.asana.com/0/488551667048375/1205823859163165/f

### Description

- Adds protections toggle to the breakage report screen
- Updates list of breakage categories to match current specification
- Adds caption below details input field 

#### Shows a caption below the breakage details form field
- [x] Go to the site breakage report screen
- [x] Verify that the "What happened? What should have happened? Did turning protections off help?" caption is displayed below the details input field.

#### Updated list of breakage categories
- [x] Go to the site breakage report screen
- [x] Tap on "Describe what happened"
- [x] Verify that the list of possible options contains these items: 
    - Site blocked or didn't load
    - Site layout broken
    - Site contains large empty spaces
    - Site asked me to disable ad blocker
    - Video didn’t play or load
    - Comments, reviews, or chats didn’t load
    - Can’t sign in/register
    - Can't pay, check out, or shop
    - Something else

Note that except for the last item, the order is randomized each time you enter that screen.

#### Shows the privacy toggle on the site breakage report screen
- [x] Go to the site breakage report screen
- [x] Verify that the protections toggle is visible above the bicycle illustration
- [x] Verify that turning protections on/off removes/adds an entry to the user_whitelist table in app.db.
- [x] Verify that the "turning protections OFF might help" is visible only if protections are ON.

#### Protections toggle on the site breakage report screen is locked to OFF when content blocking feature is disabled

- [x] Override remote config to disable the `contentBlocking` feature
- [x] Go to the site breakage report screen
- [x] Verify that the protections are OFF, and can't be turned back ON. There should be the "We temporarily turned Privacy Protections off" banner below the toggle. 

#### Protections toggle on the site breakage report screen is locked to OFF when site is in the content blocking exceptions list

- [x] Override remote config to add a site to the `contentBlocking` exceptions list (or note a site that is already there, e.g. "strava.com")
- [x] Navigate to that site
- [x] Go to the site breakage report screen
- [x] Verify that the protections are OFF, and can't be turned back ON. There should be the "We temporarily turned Privacy Protections off" banner below the toggle. 

#### Protections toggle on the site breakage report screen is locked to OFF when site is in the unprotected temporary exceptions list

- [x] Override remote config to add a site to the `unprotectedTemporary` list
- [x] Navigate to that site
- [x] Go to the site breakage report screen
- [x] Verify that the protections are OFF, and can't be turned back ON. There should be the "We temporarily turned Privacy Protections off" banner below the toggle. 

#### Verify pixels are fired

- [x] Go to the site breakage report screen
- [x] Toggle protections off/on
- [x] Verify that `m_broken_site_allowlist_add` pixel is fired on turning protections OFF
- [x] Verify that `m_broken_site_allowlist_remove` pixel is fired on turning protections ON

### UI changes
| Before | After |
| ------ | ----- |
|Screenshot_20231102_093739 | Screenshot_20231109_162714 |
|  |  Screenshot_20231109_162723 |
|  | Screenshot_20231109_162746 |
|  | Screenshot_20231109_162814 |
|  | Screenshot_20231109_162822 |

### CODE REVIEW

**Comment by lmac012** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #3774** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3774" target="_blank">Graphite</a>  👈
    * **PR #3775** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3775" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/3774?utm_source=stack-comment).

File: app/src/main/res/layout/view_site_protections_toggle.xml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+            <com.duckduckgo.mobile.android.ui.view.text.DaxTextView
+                android:id="@+id/protectionsBannerMessage"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                tools:text="@string/brokenSiteProtectionsOnBannerMessage" />
+
+        </FrameLayout>
+
+        <!-- This is just to keep the correct bottom padding when the banner is not visible -->
```
Comment (anikiki) [inline app/src/main/res/layout/view_site_protections_toggle.xml:76]: Where is this needed?
Also, do you need the outer FrameLayout?

**Review by anikiki — APPROVED** [review]
Looks good and works as expected! :tada:

File: app/src/main/res/layout/view_site_protections_toggle.xml (hunk 1 above)
Comment (lmac012) [reply app/src/main/res/layout/view_site_protections_toggle.xml]: It makes the top and bottom margins symmetrical when `protectionsBannerMessageContainer` has visibility set to `GONE`. I agree that purpose of this view is very non-obvious (which is why I added this comment), so I had a second look at this and refactored it here: https://github.com/duckduckgo/Android/pull/3774/commits/21db22d84b8c114a87afede870985f7593a18fcf

The top level FrameLayout is not needed, removed here: https://github.com/duckduckgo/Android/pull/3774/commits/5f48e0cb3d2d9ca31196e9d62686d4ea97f75a95

**Pull request merged**